app/src/main/java/com/termux/app/ExtraKeysInfos.java

```diff
@@ -0,0 +1,325 @@
+        putAll(classicArrowsDisplay);
+        putAll(wellKnownCharactersDisplay);
+        putAll(lessKnownCharactersDisplay); // NEW
+        putAll(nicerLookingDisplay);
+    }};
+
+    /**
+     * Only arrows
+     */
+    private static final CharDisplayMap arrowsOnlyCharDisplay = new CharDisplayMap() {{
+        putAll(classicArrowsDisplay);
+        // putAll(wellKnownCharactersDisplay); // REMOVED
+        // putAll(lessKnownCharactersDisplay); // REMOVED
+        putAll(nicerLookingDisplay);
+    }};
+
+    /**
+     * Full Iso
+     */
+    private static final CharDisplayMap fullIsoCharDisplay = new CharDisplayMap() {{
+        putAll(classicArrowsDisplay);
+        putAll(wellKnownCharactersDisplay);
+        putAll(lessKnownCharactersDisplay); // NEW
+        putAll(nicerLookingDisplay);
+        putAll(notKnownIsoCharacters); // NEW
+    }};
+
+    /**
+     * Some people might call our keys differently
+     */
+    static private final CharDisplayMap controlCharsAliases = new CharDisplayMap() {{
+        put("ESCAPE", "ESC");
+        put("CONTROL", "CTRL");
+        put("RETURN", "ENTER"); // Technically different keys, but most applications won't see the difference
+        put("FUNCTION", "FN");
+        // no alias for ALT
+
+        // Directions are sometimes written as first and last letter for brevety
+        put("LT", "LEFT");
+        put("RT", "RIGHT");
+        put("DN", "DOWN");
+        // put("UP", "UP"); well, "UP" is already two letters
+
+        put("PAGEUP", "PGUP");
+        put("PAGE_UP", "PGUP");
+        put("PAGE UP", "PGUP");
+        put("PAGE-UP", "PGUP");
+
+        // no alias for HOME
+        // no alias for END
+
+        put("PAGEDOWN", "PGDN");
+        put("PAGE_DOWN", "PGDN");
+        put("PAGE-DOWN", "PGDN");
+
+        put("DELETE", "DEL");
+        put("BACKSPACE", "BKSP");
+
+        // easier for writing in termux.properties
+        put("BACKSLASH", "\\");
+        put("QUOTE", "\"");
+        put("APOSTROPHE", "'");
+    }};
+
+    CharDisplayMap getSelectedCharMap() {
+        switch (style) {
+            case "arrows-only":
+                return arrowsOnlyCharDisplay;
+            case "arrows-all":
+                return lotsOfArrowsCharDisplay;
+            case "all":
+                return fullIsoCharDisplay;
+            case "none":
+                return new CharDisplayMap();
+            default:
+                return defaultCharDisplay;
+        }
+    }
+
+    /**
+     * Applies the 'controlCharsAliases' mapping to all the strings in *buttons*
+     * Modifies the array, doesn't return a new one.
+     */
+    public static String replaceAlias(String key) {
+        return controlCharsAliases.get(key, key);
+    }
+
+}
+
+class ExtraKeyButton {
+
+    /**
+     * The key that will be sent to the terminal,
+     * either a control character defined in ExtraKeysView.keyCodesForString (LEFT, RIGHT, PGUP...),
+     * either some text.
```

Comment:
Should write `or some text` here. At first, I was expecting something more after either and thought something was missing.

Reply:
Yeah this English syntax bugged me too, I wasn't sure what was the correct version though...

Reply:
This would be correct as far as I know:

> The key that will be sent to the terminal, either a control character defined in ExtraKeysView.keyCodesForString (LEFT, RIGHT, PGUP...) or some text.